dali/python/backend_impl.cc

```diff
@@ -1158,6 +1159,43 @@ PYBIND11_MODULE(backend_impl, m) {
     return context != nullptr;
   });
 
+  m.def("GetCudaVersion", [] {
+    int version = -1;
+    auto ret = cudaDriverGetVersion(&version);
+    if (ret != cudaSuccess) {
+      return -1;
+    } else {
+      return version;
+    }
+  });
+
+  m.def("GetCufftVersion", [] {
+    int ret = -1;
+    try {
+      // we don't want to throw when it is not available, just return -1
+      ret = GetCufftVersion();
+    } catch (std::runtime_error) {}
```

Comment:
```suggestion
    } catch (const std::runtime_error&) {}
```
Shouldn't exceptions be caught be `const&`? I believe having them by value here creates additional copy, but I might be wrong.

Reply:
Good catch. Fixed.